### DISCUSSION TITLE
Max model output tokens?

### DISCUSSION THREAD

**Discussion by deniseiras** [maritaca-ai/maritalk-api#108] [Q&A] [answered]
I came across a 512 token cut in the response of the sabe-3 model.
Where can I check this information for all models?
How can I change the call to return more tokens?
Cheers!

**Comment by rodrigo-f-nogueira** [comment, marked as answer]
Hi Denis, 

In the web chat, the maximum number of generated tokens is 4000.

In the API, you can configure it by setting the max_tokens parameters:
```python
import maritalk

model = maritalk.MariTalk(
    key="insira sua chave aqui. Ex: '100088...'",
    model="sabia-3"
)

response = model.generate(
    "Quanto é 25 + 27?",
    max_tokens=200)

answer = response["answer"]
```

**Reply by deniseiras** [reply]
Hi,
thanks for your answer, it works.
What is the maximum value for max_tokens ? (At https://www.maritaca.ai/ I can see "200 mil tokens de saída por minuto". So it's possible to get completion with 200k tokens in a single call ? or is there limits related to message in HTTP call ?
Cheers

**Reply by rodrigo-f-nogueira** [reply]
Hi @deniseiras,

200k tokens/min is the rate limit, i.e., if you send requests such that more than 200k tokens are generated in one minute, the server will refuse to generate more tokens. Then you have to wait one minute or so to send new requests.

The max_tokens parameter is the amount of tokens generated in a single request. For instance, if you are using Sabia-3, which supports 32000 context tokens, and your prompt (input) has 1000 tokens, max_tokens can set to a maximum of 32000 - 1000 = 31000 tokens.

**Reply by deniseiras** [reply]
Thanks Rodrigo, I understand all now!

**Reply by rodrigo-f-nogueira** [reply]
Great, please let us if you any further questions

**Reply by peq42** [reply]
> Hi Denis,
> 
> In the web chat, the maximum number of generated tokens is 4000.
> 
> In the API, you can configure it by setting the max_tokens parameters:
> 
> ```python
> import maritalk
> 
> model = maritalk.MariTalk(
>     key="insira sua chave aqui. Ex: '100088...'",
>     model="sabia-3"
> )
> 
> response = model.generate(
>     "Quanto é 25 + 27?",
>     max_tokens=200)
> 
> answer = response["answer"]
> ```

when will there be a subscription option to get more?

**Reply by rodrigo-f-nogueira** [reply]
Hi @peq42 the models now support 12,000 output tokens.
